konduit-serving-pipeline/src/main/java/ai/konduit/serving/pipeline/impl/data/ProtoData.java

```diff
@@ -0,0 +1,107 @@
+/*
+ *  ******************************************************************************
+ *  * Copyright (c) 2020 Konduit K.K.
+ *  *
+ *  * This program and the accompanying materials are made available under the
+ *  * terms of the Apache License, Version 2.0 which is available at
+ *  * https://www.apache.org/licenses/LICENSE-2.0.
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  * License for the specific language governing permissions and limitations
+ *  * under the License.
+ *  *
+ *  * SPDX-License-Identifier: Apache-2.0
+ *  *****************************************************************************
+ */
+package ai.konduit.serving.pipeline.impl.data;
+
+import ai.konduit.serving.pipeline.api.data.Data;
+import ai.konduit.serving.pipeline.impl.data.helpers.ProtobufUtils;
+import com.google.protobuf.InvalidProtocolBufferException;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.*;
+
+@Slf4j
+public class ProtoData extends JData {
+
+    public ProtoData(@NonNull InputStream stream) throws IOException {
+        this(fromStream(stream));
+    }
+
+    public ProtoData(@NonNull byte[] input) {
+        this(fromBytes(input));
+    }
+
+    public ProtoData(@NonNull File file) throws IOException {
+        this(fromFile(file));
+    }
+
+    public ProtoData(@NonNull Data data) {
+        if(data instanceof JData){
+            getDataMap().putAll(((JData)data).getDataMap());
+        }
+    }
+
+    @Override
+    public ProtoData toProtoData() {
+        return this;
+    }
+
+    @Override
+    public void save(File toFile) throws IOException {
+        write(new FileOutputStream(toFile));
```

Comment:
(a) buffer, and (b) make sure to close your streams.
Anytime you are dealing with files, you should _always_ buffer and close streams to ensure file handle is released.
```
try(OutputStream os = new BufferedOutputStream(new FileOutputStream(toFile))){
```

Reply:
Fixed.